source/io.c

```diff
@@ -867,15 +867,20 @@ PyObject *aws_py_input_stream_new(PyObject *self, PyObject *args) {
         return NULL;
     }
 
-    struct aws_input_stream *stream = aws_input_stream_new_from_py(py_self);
-    if (!stream) {
-        return PyErr_AwsLastError();
+    if (!py_self || (py_self == Py_None)) {
+        aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        return NULL;
```

Comment:
this isn't doing proper error reporting. previously, this was in a helper function that did aws_last_error() style reporting. But this is function is called directly from python and needs to raise a python exception

do something like:

```
PyErr_SetString(PyExc_TypeError, "InputStream cannot be None");
return NULL;
```